### PR TITLE
Toggle UI: reuse existing component for Content Delivery page (#8829)

### LINKED ISSUE #8829: Toggle UI: reuse existing component (Addons/Analytics) as interim solution for Content Delivery page
## Context

Sub-task of #8818 (Toggle & notice UI components for Content Delivery page), part of the RocketCDN Free Tiers refactor epic (#8693).

#8818 specs a final toggle component matching the Figma design. As a first step to unblock Story 2 (#8700) faster, this issue proposes reusing our **existing toggle component** (the one already used on the Addons page and for the Analytics opt-in) instead of building the Figma-specific toggle right away.

## Scope

- Use the current/existing toggle UI component (as used in the Addons page and Analytics settings) for the CDN mode switch, reusable across the three CDN mode tabs (RocketCDN Free, RocketCDN Pro, Other CDN).
- Cover the on / off / disabled-forced-off (with tooltip) states with this existing component.
- Notice/banner variants and matching the final Figma toggle design remain covered by #8818 / deferred to a later pass, as already scoped there.

## Out of scope

- Final visual implementation matching the Figma design for the toggle — this is an interim step to speed up delivery of #8700; polishing the toggle to match Figma is deferred to a later pass under #8818.
- Notice/banner implementations (covered by each mode's own story per the epic).

## Related

- Parent: #8818
- Epic: #8693
- Story 2 — Toggle-driven Content Delivery UI: #8700

### PR DESCRIPTION
**Pull request by hellofromahmed**
# Description

Fixes #8829
As an interim step to unblock Story 2 (#8700) faster, this replaces the custom checkbox-slider toggle on the Content Delivery page's three CDN mode sections (RocketCDN Free, RocketCDN Paid, Other CDN) with the plugin's existing toggle component (already used on the Addons page and for the Rocket Analytics opt-in), and moves the toggle to sit right after the section title instead of before it. Users see a consistent, styled ON/OFF switch instead of a plain unstyled checkbox, with no change to the underlying CDN mode-switching behavior.

## Type of change

- [ ] New feature (non-breaking change which adds functionality).
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] Enhancement (non-breaking change which improves an existing functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as before).
- [x] Sub-task of #8818
- [ ] Chore
- [ ] Release

### What was tested

Manual, in a local WP environment with WP Rocket active:
- Loaded Settings > WP Rocket > Content Delivery, RocketCDN tab: confirmed the toggle for RocketCDN Free (and RocketCDN Paid, when that section is applicable) renders as the styled pill ON/OFF switch immediately after the section title, vertically centered with it.
- Loaded the Other CDN tab: confirmed the "Your CDN" toggle renders the same way.
- Clicked each toggle: confirmed mode switching remains mutually exclusive (checking one unchecks the others), the REST call to `/wp-rocket/v1/rocketcdn/mode` still fires, the active tab/indicator updates, and no JS console errors appear.
- Verified the forced-off (host-managed CDN) state still disables the toggle and now shows an explanatory tooltip on hover.
- Confirmed `php -l` and `vendor/bin/phpcs` are clean on all touched/new view files.

https://github.com/user-attachments/assets/57a8a6b9-9e47-4a7d-bc77-c38c1350057e

### How to test

1. Check out this branch on a WP install with WP Rocket active.
2. Go to Settings > WP Rocket > Content Delivery.
3. On the RocketCDN tab, verify the toggle for RocketCDN Free (or RocketCDN Paid, if the site is on the paid plan) sits immediately after the title/badge, is vertically aligned with it, and toggling it switches the active CDN mode.
4. Switch to the Other CDN tab and verify the same for the "Your CDN" toggle.
5. If your environment has a way to force `is_forced_off` (e.g. the OneCom hosting compatibility layer), confirm the toggle is disabled and hovering it shows the "managed by your host" tooltip.

### Affected Features & Quality Assurance Scope

- Content Delivery settings page (Settings > WP Rocket > Content Delivery): RocketCDN Free section, RocketCDN Paid section, Other CDN (BYOCDN) section.
- No changes to the CDN mode-switching logic, REST endpoint, or JS (`src/js/global/cdn-driver.js` is untouched); only the toggle's markup/CSS and the newly shared partial that renders it.

### Documentation

- Added `views/settings/partials/cdn/cdn-mode-toggle.php`, a shared partial rendered via the existing `render_parts_with_data( 'cdn/cdn-mode-toggle', $data )` helper (same pattern already used for `cdn/cdn-status-indicator`). It outputs the plugin's existing `wpr-radio` / `wpr-radio-ui` toggle markup (the component used on the Addons page and for the Rocket Analytics opt-in), keeping the `wpr-cdn-mode-toggle__input` class and `data-cdn-mode` attribute the existing JS depends on.
- `views/settings/sections/cdn/rocketcdn-free.php`, `rocketcdn-paid.php`, and `your-own-cdn.php` now call this partial instead of duplicating the toggle markup, and nest it inside `.wpr-optionHeader__title-group` (right after the title/badge) instead of rendering it before the title.
- `src/scss/components/_contentDelivery.scss`: added `.wpr-cdn-mode-toggle` styles (hover-revealed tooltip for the forced-off state, and an explicit height on the wrapper — since this toggle's label carries no visible text, it was collapsing to 0 height and being mis-centered by flexbox).
- Rebuilt `assets/css/wpr-admin.css` / `.min.css` via `npm run build:css`.

### New dependencies

None.

### Risks

Low. This only changes markup/CSS for an interim UI (final Figma-matched toggle design is deferred to #8818) and does not touch the CDN mode-switching JS, REST endpoint, or PHP controller logic. The toggle's input `class` and `data-cdn-mode` attribute — which the existing JS selects on — are unchanged.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

## Unticked items justification

No new automated tests were added: this is a markup/CSS-only change to an interim UI (the toggle's visual design itself is still slated to change under #8818), and no existing test asserts on this markup, so a snapshot/DOM test would only pin down implementation details that are expected to change again shortly.

# Additional Checks
- [ ] In the case of complex code, I wrote comments to explain it.
- [ ] When possible, I prepared ways to observe the implemented system (logs, data, etc.).
- [ ] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

### CODE REVIEW

**Comment by hellofromahmed** [comment]
Fixed a bug found while testing the RocketCDN Free toggle on this branch: enabling it saved correctly to the DB (`cdn` / `cdn_type` / `cdn_state` all persisted as expected), but the toggle kept rendering **OFF** on page load.

**Root cause:** `CdnStateBridge::resolve_live()` is hooked on `pre_get_rocket_option_cdn_state` and always returned a non-`null` string, which unconditionally short-circuits `Options_Data::get()` — so the real, persisted `cdn_state` value was never actually read. Instead it was recomputed live via `legacy_to_state()`. When no RocketCDN token/subscription exists yet (true for Free-tier activation), `APIClient::get_remote_subscription_data()` returns its default with `subscription_status => 'cancelled'`, which made `is_cancelled_outside_grace_period()` return `true` and forced `legacy_to_state()` to collapse to `CDN_STATE_NOTHING` — overriding the correct `rocketcdn_free` state on every read regardless of `cdn`/`cdn_type` being set correctly.

**Fix pushed in 5254d96fd:**
- `CdnStateBridge::resolve_live()` / `legacy_to_state()` now return `null` (instead of forcing `CDN_STATE_NOTHING`) in the cancelled/no-token case, so the filter no longer short-circuits and the persisted `cdn_state` value flows through normally.
- `Render\Controller::is_cdn_paused()` now derives paused state from `get_applied_cdn_state()` instead of the raw `cdn` option, and the RocketCDN paid status text is only shown as paused when there's an active subscription.
- `Rest::save_cdn_mode()` now returns the full pages payload (`items_html`/`status_indicator_html`/etc.) alongside the mode-change response, so the toggle handler can refresh those UI pieces immediately.
- `cdn-driver.js`: consolidated the post-response subscription-loading/status-indicator refresh into a shared `refreshUIElements()`, wired into the toggle handler as well.
- `ajax.js`: excluded the RocketCDN free/BYOCDN toggles from the generic checkbox autosave handler since `cdn-driver.js` already owns their save flow.

**Comment by hellofromahmed** [comment]
CI on the previous push (5254d96fd) was red on all WP/PHP unit-test matrix jobs due to one pre-existing test: `Test_AddExcludeCdnSection::testShouldReturnCorrectHelpUrl` (data set `testWhenRocketCdnActiveInitialUrlIsRocketCdnUrl`) still stubbed `Options_Data::get('cdn')`, but `is_cdn_paused()` was changed in this branch to call `Context::get_applied_cdn_state()` instead — so Mockery threw `BadMethodCallException: Received ... get_applied_cdn_state(), but no expectations were specified`.

Fixed in cec91f6c5 by updating the test's mock to match the new dependency. Full local unit suite (2814 tests) passes clean now.

**Comment by hellofromahmed** [comment]
While testing this branch's toggle work I noticed #8804 (fix/8704-license-expiry-handling-for) independently built a "why is this toggle disabled" tooltip on the same three CDN view templates and `Render\Controller`, forked from the same base commit (`535a53652`) before this branch's toggle-reuse refactor existed. Landing both as-is would conflict: #8804 reintroduces the old triplicated inline `<label class="wpr-cdn-mode-toggle">` markup (this branch already dedup'd that into `views/settings/partials/cdn/cdn-mode-toggle.php`) plus a new `forced_off_tooltip` section key, alongside this branch's existing `toggle_tooltip` key - and neither tooltip covers the other's reasons (ours: hosting-compat only; theirs: subscription-loading/expired-licence/banned-reseller only).

Folded #8804's reasons into this branch's existing tooltip in 597c1d659 instead of taking its view/markup changes:
- `should_reject_rocketcdn_activation()` gains the banned-reseller term (same as #8804, same rationale for why it lives here and not in `should_display_licence_expired_notice()` or `is_forced_paused()`).
- New `get_rocketcdn_toggle_forced_off_tooltip()` covers subscription-loading / expired-licence / banned-reseller copy, wired into `toggle_tooltip` for the RocketCDN Free and Paid sections only.
- `get_toggle_forced_off_tooltip()` (hosting-compatibility message) stays wired to the BYOCDN section unchanged - that section's `is_forced_off` is never set for licence/subscription reasons.
- Added `shouldRejectRocketcdnActivation.php` and `getRocketcdnToggleForcedOffTooltip.php` unit tests, patched the two RocketCDN section-builder tests for the new `is_reseller_license_banned()` dependency.

Left the same note on #8804 so whoever merges it into the shared base knows to drop its view/markup + `forced_off_tooltip` changes in favor of what's here, rather than resolving it as a textual merge.

**Review by jeawhanlee — COMMENTED** [review]
I left some comment on https://github.com/wp-media/wp-rocket/pull/8804 about using `is_forced_paused()` instead of `should_reject_rocketcdn_activation` which I see similarities here, probably the branch was merged here, If it's resolved there, this can be ported here if there's a necessary change.

**Comment by Mai-Saad** [comment]
@hellofromahmed Thanks for the PR, Please find possible risk shared by AI and confirm if we need any changes :pray: 

> src/js/global/ajax.js has a generic handler bound to .wpr-radio input[type=checkbox] that fires a rocket_toggle_option AJAX save on change, with an exclusion list for elements handled elsewhere. Before this PR, the three CDN toggles used bespoke markup (<label class="wpr-cdn-mode-toggle">) that never matched .wpr-radio, so this handler never fired for them.
> 
> This PR reuses the shared wpr-radio component (views/settings/partials/cdn/cdn-mode-toggle.php:171 wraps the input in <div class="wpr-radio wpr-cdn-mode-toggle">), so all three toggles now match that selector for the first time. The PR updates the exclusion list but only adds two of the three IDs:
> 
> 
> var excluded = [ 'cloudflare_auto_settings', 'cloudflare_devmode', 'analytics_enabled', 'wpr-rocketcdn-free-toggle', 'wpr-byocdn-toggle' ];
> wpr-rocketcdn-paid-toggle is missing. So toggling RocketCDN Paid now fires an extra, unintended rocket_toggle_option AJAX POST alongside the real /wp-rocket/v1/rocketcdn/mode REST call. I checked the server-side handler (Page::toggle_option() in inc/Engine/Admin/Settings/Page.php) — it whitelists allowed option names and wpr-rocketcdn-paid-toggle isn't in that list, so today it's harmless (just an extra request that wp_die()s). But it's clearly an unintentional omission — a stray failed/no-op request on every Paid-toggle click, and a landmine if that whitelist is ever extended later without anyone remembering this coupling.
> 
> Fix: add 'wpr-rocketcdn-paid-toggle' to the excluded array in src/js/global/ajax.js (around line 67).
